lib/performify/base.rb

```diff
@@ -58,8 +58,9 @@ def success!(with_callbacks: true)
       @result
     end
 
-    def fail!(with_callbacks: true)
+    def fail!(with_callbacks: true, errors: nil)
       @result = false
+      errors!(errors) unless errors.nil?
```

Comment:
maybe need to add check that `errors` is a Hash? as I know `errors!(...)` will merge this hash with existing errors